src/Tizen.Security.SecureRepository/Tizen.Security.SecureRepository/DataManager.cs

```diff
@@ -55,15 +55,53 @@ static public byte[] Get(string alias, string password)
 
             try
             {
-                Interop.CheckNThrowException(
-                    Interop.CkmcManager.GetData(alias, password, out ptr),
+                CheckNThrowException(
+                    CkmcManager.GetData(alias, password, out ptr),
                     "Failed to get certificate. alias=" + alias);
                 return new SafeRawBufferHandle(ptr).Data;
             }
             finally
             {
                 if (ptr != IntPtr.Zero)
-                    Interop.CkmcTypes.BufferFree(ptr);
+                    CkmcTypes.BufferFree(ptr);
+            }
+        }
+
+        /// <summary>
+        /// Checks for alias existence
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        /// <param name="alias">The name of a certificate to retrieve.</param>
+        /// <returns>Boolean indicating the exitsance of the alias.</returns>
+        /// <exception cref="ArgumentNullException">
+        /// The alias argument is null.
+        /// </exception>
+        /// <exception cref="InvalidOperationException">
+        /// Indicates failure to communicate with the Keystore. Check the errorcode for details
+        /// </exception>
+        static public bool AliasExists(string alias)
+        {
+            if (alias == null)
+                throw new ArgumentNullException("alias cannot be null");
+
+            IntPtr ptr = IntPtr.Zero;
+
+            try
+            {
+                var errorCode = CkmcManager.GetData(alias, string.Empty, out ptr);
+                if((int)KeyManagerError.UnknownAlias == errorCode)
+                    return false;
+
+                if((int)KeyManagerError.AuthenticationFailed == errorCode ||
```

Comment:
Why do you determine as existed when Authentication was failed

Reply:
I see, AuthenticationFailed could be riase when only key was existed

Reply:
AuthenticationFailed will be thrown if the alias exists but we did not supply the correct password.